File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def in_trav(self, root, vec):
        if root is None:
            return
        self.in_trav(root.left, vec)
        vec.append(root.val)
        self.in_trav(root.right, vec)

    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        res = []
        self.in_trav(root, res)
        return res

        

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void in_trav(TreeNode* root, vector<int>& vec){
        if (root == nullptr)
            return;
        in_trav(root->left, vec);
        vec.push_back(root->val);
        in_trav(root->right, vec);
    }
    
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> result;
        in_trav(root, result);
        return result;
    }
};